src/controller/subtitle-stream-controller.js

```diff
@@ -245,6 +261,10 @@ export class SubtitleStreamController extends BaseStreamController {
   }
 
   onMediaSeeking () {
-    this.fragPrevious = null;
+    this.stopLoad();
+  }
+
+  onMediaSeeked () {
+    this.startLoad();
```

Comment:
🤔 I don't think we should ever call `stopLoad` or `startLoad` based on video events. If you stopped loading, but then set `video.currentTime`, as a result, subtitles would start loading here when `onMediaSeeked` is envoked.

Could we simply set `this.fragPrevious = null` in `onMediaSeeking`? ...and call `tick()` if you want speed things up, like this:
```suggestion
    this.fragPrevious = null;
    this.tick(); // or super.onMediaSeeking();
```
Keep in mind you are overriding [base-stream-controller `onMediaSeeking`](https://github.com/video-dev/hls.js/blob/45436cdaf7c118411bd5a271ffb345c2bbfe3380/src/controller/base-stream-controller.js#L40) here which is supposed to clear fragPrevious when necessary, but probably not working with this subtitle controller's fake buffer.

I think `currentTime` is updated on `'seeking'`, so we don't need to wait until `'seeked'` to load. You want subtitles to be requested as `currentTime` jumps in `'seeking'`.

Reply:
Hi @robwalch ,
thanks for reviewing! I did some refactoring based on your hints and was able to remove the dependency of the onMediaSeeked-Event. I was not aware that the `currentTime` is already set on `'seeking'`.
Setting `this.fragPrevious = null` in `onMediaSeeking` (with or without a combinded `tick()` call) is does not solve the problem. We actually need to cancel the existing loaders as otherwise, as soon as the loader finishes loading, it sets back the `this.fragPrevious` var inside the `onSubtitleFragProcessed`. That is why i (for simplicy) called `this.stopLoad()`. 
But i agree that it is nicer to solve it by simply adding some more disposement logic in the `onMediaSeeking` event:
1) stop the current loader
2) unset the `this.fragCurrent` and `this.fragPrevious` variables
3) reset the state to `State.IDLE` so the next tick is able to start loading subtitle segments
4) do a manual tick (just to be a little quicker)

I was not able to reproduce the issue with this changes: https://github.com/video-dev/hls.js/pull/3063/commits/991de69d212c974486bb16a399c1b17d4d0fad47